job-service-db/src/main/resources/procedures/getJobExists.sql

```diff
@@ -0,0 +1,41 @@
+--
+-- Copyright 2015-2018 Micro Focus or one of its affiliates.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+/*
+ *  Name: get_job_exists
+ *
+ *  Description:
+ *  Returns a record indicating whether the job has a specific value by comparing the hash.  Returns
+ *  0 records when the job doesn't exist.
+ */
+CREATE OR REPLACE FUNCTION get_job_exists(
+    in_partition VARCHAR(40),
+    in_job_id VARCHAR(48),
+    in_job_hash INT
+)
+RETURNS TABLE(
+    job_exists BOOLEAN
+)
+LANGUAGE plpgsql STABLE
+AS $$
+BEGIN
+    RETURN QUERY
+    SELECT job.job_hash = in_job_hash
+    FROM job
+    WHERE job.partition = in_partition
+        AND job.job_id = in_job_id;
+END
+$$;
```

Comment:
There's something a bit odd about this:
- It returns no rows if the job doesn't exist at all.
- It returns one row containing false if the job exists but the hash doesn't match.
- It returns one row containing true if the job exists but that hash does match.

Perhaps a `get_job_hash` function which just returned the hash would be cleaner?  Or use `EXISTS` so that it can be relied upon to return one row with just `true` or `false`.

Reply:
In an old version I wanted to differentiate between the 3 cases, but I guess I don't any more.  I'll make it always return 1 record.